api/turing/service/ensembling_job_service.go

```diff
@@ -29,14 +29,16 @@ type EnsemblingJobListOptions struct {
 
 // EnsemblingJobService is the data access object for the EnsemblingJob from the db.
 type EnsemblingJobService interface {
-	Save(ensembler *models.EnsemblingJob) error
+	Save(ensemblingJob *models.EnsemblingJob) error
+	Delete(ensemblingJob *models.EnsemblingJob) error
 	FindByID(id models.ID, options EnsemblingJobFindByIDOptions) (*models.EnsemblingJob, error)
 	List(options EnsemblingJobListOptions) (*PaginatedResults, error)
 	CreateEnsemblingJob(
 		job *models.EnsemblingJob,
 		projectID models.ID,
 		ensembler *models.PyFuncEnsembler,
 	) (*models.EnsemblingJob, error)
+	DeleteEnsemblingJob(ensemblingJob *models.EnsemblingJob) error
```

Comment:
I would like to see the DAO service and orchestration service to be split into two. It's confusing to have `Delete` and `DeleteEnsemblingJob` in a single service

Reply:
Now, after inspecting `DeleteEnsemblingJob` implementation, I think that maybe it should be renamed accordingly to what it's doing. Maybe `MarkJobForTermination`?

Reply:
Ya actually MarkJobForTermination is much better.